Classes_Homework_5/fromfile.py

```diff
@@ -0,0 +1,181 @@
+        return value.strip()
+
+    def check_value(self, label, my_str, start):
+        value = self.get_value(my_str, start)
+        if value:
+            return value
+        else:
+            raise NoValue (label, start)
+
+    def parse_file(self, file_text):
+        sections = self.get_sections(file_text)  # split text into sections (by Section keyword)
+        final_text = ''
+        for section in sections:
+            com = Combine(section.split()[0], decor, decor_length)
+            if section.split()[0] == label1:
+                news_city = self.check_value(label1, section, city+':')
+                news_text = self.check_value(label1, section, text+':')
+                final_text += com.get_news(news_city, news_text)+'\n\n'
+            elif section.split()[0] == label2:
+                ad_text = self.check_value(label2, section, text+':')
+                # convert date passed as string to datetime.date
+                ad_date = d.str_to_date(self.check_value(label2, section, date+':'), cnf.get_values("PATTERNS", "date_format"))
+                # check if date is not in the past
+                di.raise_if_past(ad_date)
+                final_text += com.get_ad(ad_text, ad_date)+'\n\n'
+            elif section.split()[0] == label3:
+                rec_text = self.check_value(label3, section, text+':')
+                rec_calories = self.check_value(label3, section, calories+':')
+                final_text += com.get_recipe(rec_text, rec_calories)+'\n\n'
+        return final_text.rstrip()
+
+    def raise_if_empty(self, my_str):
+        """Verify the string is not empty"""
+        if not my_str:
+            raise NoSectionsError(section)
+
+    def verify_source(self, source):
+        """Open and try to parse file(s), catch errors, return file path and text"""
+        try:
+            with fileinput.input(files=source) as fp:
+                file_text = []
+                for line in fp:
+                    file_text.append(line.strip())
+            parsed_text = self.parse_file('\n'.join(file_text))
+            self.raise_if_empty(parsed_text)
+            return source, parsed_text
+        except UnicodeDecodeError:
+            print(cnf.get_values("ERRORS", "cannot_read")+"\n")
+            return None, None
+        except (OSError, ValueError, NoSectionsError, NoValue, PastDate) as err:
+            print(err)
+            return None, None
+
+    def read_source(self, raw_source):
+        """If source file returns no errors, return file path and text, else - ask user to input path to source file"""
+        proper_source, result = self.verify_source(raw_source)  # check path passed by user/default path first
+        inp_request = cnf.get_values("INPUTS", "filepath")
+        while not proper_source:  # if any errors occur with initial source, ask user to input another path as long as it doesn't meet the requirements
```

Comment:
Try to avoid so long lines. By the way, the recommendation from PEP8 is max 80 characters.